src/clients/java/src/client.zig

```diff
@@ -708,3 +742,174 @@ const JNIHelper = struct {
         return @ptrCast(address[0..@as(usize, @intCast(length))]);
     }
 };
+
+/// Helper for managing the `AttachCurrentThread`/`DetachCurrentThread` lifecycle
+/// when the JNI layer is unaware of when the native thread exits.
+/// https://developer.android.com/training/articles/perf-jni#threads
+const JNIThreadHelper = struct {
+    var tls_key: tls.Key = undefined;
+    var create_key_once = std.once(create_key);
+
+    /// This function calls `AttachCurrentThreadAsDaemon` to attach the current native thread to
+    /// the JVM as a daemon thread. It also registers a callback to call `DetachCurrentThread`
+    /// when the thread exits (e.g., when the client is closed or evicted).
+    pub fn attach_current_thread_with_cleanup(jvm: *jni.JavaVM) *jni.JNIEnv {
+        // Create the tls key once per JVM.
+        create_key_once.call();
+
+        // Set the JVM handler to the thread-local storage slot for each time a native
+        // thread is started.
+        tls.set_key(tls_key, jvm);
+
+        return JNIHelper.attach_current_thread(jvm);
+    }
+
+    /// Create the thread-local storage key and the corresponding destructor callback.
+    /// Note: We don't need to delete the key because the JNI module cannot be unloaded,
+    /// so it will always be available for the duration of the JVM process.
+    fn create_key() void {
+        tls_key = tls.create_key(&destructor_callback);
+    }
+
+    // Will be called by the OS with the JVM handler when the thread finalizes.
+    fn destructor_callback(value: *anyopaque) callconv(.C) void {
+        JNIHelper.detach_current_thread(@ptrCast(value));
+    }
```

Comment:
Can we give this a more descriptive name (`jvm`)?

Reply:
It can't be `jvm` because the unit test uses another pointer type.
But I think we can call it `tls_value` since it's indexed by the `tls_key`.

Reply:
The test uses its own `destructor_callback` implementation, so we could name it more descriptively here, right?